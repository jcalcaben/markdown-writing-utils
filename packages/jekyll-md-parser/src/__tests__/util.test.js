const {looksLikeDefinition, containsLiquid} = require('../util')

test("Normal definition", ()=>{
    const input = "[link definition]: https://www.google.com"

    const matchDefinition = looksLikeDefinition(input);
    const matchLiquid = containsLiquid(input);

    expect(matchDefinition).toMatchSnapshot();
    expect(matchLiquid).toBeFalsy();
})

test("Not a definition link", ()=>{
    const input = "{not a link definition}: Foo"

    const match = looksLikeDefinition(input);

    expect(match).toBeFalsy();
})

test("Jekyll link", ()=>{
    const input = "[jekyll link]: {%link file/location/index.md %}"

    const matchDefinition = looksLikeDefinition(input);
    const matchLiquid = containsLiquid(input);

    expect(matchDefinition).toMatchSnapshot();
    expect(matchLiquid).toMatchSnapshot();
})

test("Jekyll variable", ()=>{
    const input = "[jekyll variable]: {{site.baseurl}}/some/url/location"

    const matchDefinition = looksLikeDefinition(input);
    const matchLiquid = containsLiquid(input);

    expect(matchDefinition).toMatchSnapshot();
    expect(matchLiquid).toMatchSnapshot();
})