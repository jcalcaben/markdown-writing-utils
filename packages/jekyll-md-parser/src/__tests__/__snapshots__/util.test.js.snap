// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jekyll link 1`] = `
Array [
  "[jekyll link]: ",
  "jekyll link",
]
`;

exports[`Jekyll link 2`] = `
Array [
  "{%link file/location/index.md %}",
  "%link file/location/index.md %",
]
`;

exports[`Jekyll link 3`] = `
Array [
  "{%link file/location/index.md %}",
  "file/location/index.md ",
]
`;

exports[`Jekyll variable 1`] = `
Array [
  "[jekyll variable]: ",
  "jekyll variable",
]
`;

exports[`Jekyll variable 2`] = `
Array [
  "{site.baseurl}",
  "site.baseurl",
]
`;

exports[`Jekyll variable 3`] = `
Array [
  "{{site.baseurl}}",
  "site.baseurl",
]
`;

exports[`Normal definition 1`] = `
Array [
  "[link definition]: ",
  "link definition",
]
`;
